src/Algebra/Graph/Internal.hs

```diff
@@ -108,14 +108,7 @@ connectFoci x y = Focus (ok x || ok y) (xs <> is y) (os x <> ys) (vs x <> vs y)
 -- | A safe version of 'foldr1'.
 foldr1Safe :: (a -> a -> a) -> [a] -> Maybe a
```

Comment:
I think this function can actually be called `sconcat`, i.e. concatenating a list of semigroup values :)

Reply:
Indeed :) Do you not fear a name clash with the actual `Data.Semigroup.sconcat` ?

Reply:
Ah, good point. We do often import `Data.Semigroup`, so the name clash may become annoying. OK, let's keep `foldr1Safe` for now.